apps/www/src/contexts/UserContext.tsx

```diff
@@ -140,36 +141,57 @@ export const UserProvider = ({ children }: { children: ReactNode }) => {
   }, [providerContext]);
 
   const login = async () => {
-    const { wallet } = await connect({
-      modalMode: "canAsk",
-      provider: providerContext?.provider,
+    const res = await connect({
+      modalMode: "alwaysAsk",
+      modalTheme: "dark",
+      dappName: "Strapex",
+      resultType: "wallet",
     });
 
-    if (wallet && wallet.isConnected) {
-      const response = await fetch("/api/generateJWToken", {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify({ walletAddress: wallet.selectedAddress }),
-      });
-
-      if (!response.ok) {
-        throw new Error("Failed to generate JWT token");
-      }
+    const { wallet, connector, connectorData } = res
+    console.log(wallet)
+    console.log(connector)
+    console.log(connectorData)
+    console.log(connectorData?.account)
+    console.log(typeof connectorData?.account)
+
+    const chainId = await wallet?.request({
+      type: "wallet_requestChainId"
+    })
 
-      const { token } = await response.json();
+    console.log(chainId)
+
+    if (!wallet || !connector || !connectorData) {
+      console.log('No wallet, connector or connectorData')
+      return null
+    }
 
-      await app.logIn(Realm.Credentials.jwt(token));
-      const userId = app.currentUser?.id;
-      console.log("User logged in: ", userId);
+    const response = await fetch("/api/generateJWToken", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      // Braavos wallet returns a value of undefined for selectedAddress property of a wallet, so it is a problem
+      body: JSON.stringify({ walletAddress: connectorData?.account }),
+    });
 
-      setConnection(wallet);
-      setAccount(wallet.account);
-      setAddress(wallet.selectedAddress);
-      setUserId(userId ?? null);
-      setIsLoggedIn(true);
+    if (!response.ok) {
+      throw new Error("Failed to generate JWT token");
     }
+
+    const { token } = await response.json();
+
+    await app.logIn(Realm.Credentials.jwt(token));
+    const userId = app.currentUser?.id;
+    console.log("User logged in: ", userId);
+
+    setConnection(wallet);
+    // @ts-expect-error this is still what is gotten for the time being
```

Comment:
Can we remove this if this is not being used
```suggestion
    // @ts-expect-error this is still what is gotten for the time being
```